File: Cpp/1491_average_salary_excluding_the_minimum_and_maximum_salary.cpp
// https://leetcode.cn/problems/average-salary-excluding-the-minimum-and-maximum-salary/

class Solution {
public:
    double average(vector<int>& salary) {
        int min_ = salary[0], max_ = salary[0], s = 0;
        for (auto val: salary) {
            min_ = min(min_, val);
            max_ = max(max_, val);
            s += val;
        }
        double res = (s - min_ - max_) / (salary.size() - 2);
        return res;
    }
};

File: Python3/1491_average_salary_excluding_the_minimum_and_maximum_salary.py
# https://leetcode.cn/problems/average-salary-excluding-the-minimum-and-maximum-salary/

class Solution:
    def average(self, salary):
        """
        Inputs:
            salary: list[int]
        Outputs:
            res: float
        """
        min_, max_ = salary[0], salary[0]
        s = 0.
        for val in salary:
            min_ = min(min_, val)
            max_ = max(max_, val)
            s += val
        res = (s - min_ - max_) / (len(salary) - 2)
        return res